maint/extractcvars.in

```diff
@@ -331,27 +335,44 @@ foreach my $p (@cvars) {
     push @env_names, "${alt_ns}_" . $p->{name};
     push @env_names, "${uc_ns}_" . $p->{name};
 
+    print OUTPUT_C "    got_rc = 0;\n";
     foreach my $env_name (@env_names) {
         # assumes rc is defined
         if ($p->{type} eq 'range') {
 print OUTPUT_C <<EOT;
     rc = MPL_env2${env_fn}("$env_name", &($var_name.low), &($var_name.high));
     MPIR_ERR_CHKANDJUMP1((-1 == rc),mpi_errno,MPI_ERR_OTHER,"**envvarparse","**envvarparse %s","$env_name");
+    got_rc += rc;
 EOT
         }
         elsif ($p->{type} eq 'string' or $p->{type} eq 'enum') {
 print OUTPUT_C <<EOT;
     rc = MPL_env2${env_fn}("$env_name", &tmp_str);
     MPIR_ERR_CHKANDJUMP1((-1 == rc),mpi_errno,MPI_ERR_OTHER,"**envvarparse","**envvarparse %s","$env_name");
+    got_rc += rc;
 EOT
         }
         else {
 print OUTPUT_C <<EOT;
     rc = MPL_env2${env_fn}("$env_name", &($var_name));
     MPIR_ERR_CHKANDJUMP1((-1 == rc),mpi_errno,MPI_ERR_OTHER,"**envvarparse","**envvarparse %s","$env_name");
+    got_rc += rc;
 EOT
         }
     }
+
+    # debugging
+    print OUTPUT_C "    if (got_rc && debug) {\n";
```

Comment:
So the idea with `got_rc` is that if all variables are unset (rc=0), then nothing will be printed out? But if there is _any_ CVAR set in the env, and there wasn't an error, then the debug output will be printed?

Reply:
Yes.